optimum/habana/transformers/generation/utils.py

```diff
@@ -331,11 +461,10 @@ def generate(
         if generation_config.static_shapes is None:
             generation_config.static_shapes = self.config.model_type in MODELS_OPTIMIZED_WITH_STATIC_SHAPES
         if generation_config.ignore_eos is None:
-            generation_config.ignore_eos = lazy_mode
+            generation_config.ignore_eos = not lazy_mode
```

Comment:
I don't understand this. `ignore_eos` is used to generate tokens till `max_length` is reached, so this should be done in lazy mode. Why inverting it?

Reply:
What should be the default behavior? 
I inverted it as I was checking the perf and in earlier case when ignore_eos wasn't there in generation config. Ignore_eos is False.

Reply:
Ah okay I see. So, now, `ignore_eos` should be initialized to `None` in the generation config if not specified, and here we should set it to the same value as `lazy_mode` because in lazy mode we don't want to stop when sequences are finished.
This was actually the reason why regression tests failed. Setting it back to `lazy_mode` gives the expected numbers.